math/divisor.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file
+ * @brief Algorithms to find divisor of all natural numbers.
+ * @author [Tee Chia Suen](https://github.com/ChiaSuen)
+ */
+
+#include <cassert>  /// for assert
+#include <iostream>
+#include <vector>  //for declaring undefined size array
+
+/**
+ * @namespace <check from other files in this repo>
+ */
+namespace math {
+
+/**
+ * @brief Function to calculate the divisors and append it to a vector array
+ * @tparam int
+ * @param Number to calculate the divisors
+ */
+template <typename T>
+std::vector<int> Divisors(T n) {
+    int divisor = 0;            // divisor
+    std::vector<int> divisors;  // divisor array
+
+    for (divisor = 1; divisor * divisor < n; divisor++) {
+        if (n % divisor == 0) {
+            divisors.push_back(divisor);
+        }
+    }
+    if (divisor - (n / divisor) == 1) {
+        divisor--;
+    }
+    for (int arr_size = int(divisors.size()) - 1; arr_size >= 0; arr_size--) {
+        if (n % divisors[arr_size] == 0) {
+            divisors.push_back(n / divisors[arr_size]);
+        }
+    }
+
+    return divisors;
+}
+
+/**
+ * @brief Function to print vector array of divisors
+ * @tparam std::vector< int >
```

Comment:
Don't forget to add a detailed description for this parameter here as well.
```suggestion
 * @tparam T
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix